golem-shard-manager/src/model.rs

```diff
@@ -79,11 +78,30 @@ impl Pod {
             ip: source_ip,
         };
 
-        let resolved = pod.address()?.map(|sa| sa.ip()).collect::<Vec<_>>();
-        if !resolved.contains(&source_ip) {
-            warn!("Host mismatch between registration message and resolved message source. Provided: {resolved:?}; source: {source_ip:?}");
+        match pod.address() {
+            Ok(resolved_addresses) => {
+                let resolved_addresses = resolved_addresses.map(|sa| sa.ip()).collect::<Vec<_>>();
+
+                if !resolved_addresses.contains(&source_ip) {
+                    warn!(
+                        pod= ?pod,
+                        resolved_adresses = ?resolved_addresses,
+                        source_ip = ?source_ip,
```

Comment:
Similar here and in some other cases, can we use Display instead of Debug ?

Reply:
https://github.com/golemcloud/golem/pull/690/commits/56007858afaaa4410dfdd57c513c330b9327786a

i searched for `= ?`, did not found more

Reply:
thanks for the other comments, missed those